app/src/main/java/com/example/cinemapp/util/mappers/ActorDetailsMapper.kt

```diff
@@ -0,0 +1,61 @@
+package com.example.cinemapp.util.mappers
+
+import com.example.cinemapp.data.CastMovieCreditDTO
+import com.example.cinemapp.data.PersonDetailsDTO
+import com.example.cinemapp.ui.main.model.CastMovieCredit
+import com.example.cinemapp.ui.main.model.PersonDetails
+import java.time.LocalDate
+
+class ActorDetailsMapper(
+    private val urlMapper: URLMapper
+) {
+    fun mapDateStringListToDescendingYearsList(dates: List<String?>?): List<Int?> {
+        return dates?.map { date ->
+            if (date.isNullOrEmpty()) null
+            else date.substring(0, 4).toInt()
+        }?.distinct()?.sortedByDescending { year -> year ?: 0 } ?: emptyList()
+    }
+
+    private fun mapGenderIdToString(gender: Int): String {
+        return when (gender) {
+            1 -> "Female"
+            2 -> "Male"
+            3 -> "Non-binary"
+            else -> "Not specified"
+        }
+    }
+
+    fun mapPersonDetailsDTOToPersonDetails(
```

Comment:
`mapToDetails` maybe?

Reply:
I would rather the first choice since the second would not be possible for functions that take and return a list. For Kotlin all of those would have the same signature so 2 of those list mapping functions wouldn't be able to be in the same class with the same name.

Reply:
As you wish 👍